crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs

```diff
@@ -1,60 +1,55 @@
-use super::{
-    dsa::ed25519::{Ed25519Generator, Ed25519Signer},
-    jwk::{Jwk, JwkError},
+use super::{key_manager::KeyManager, KeyManagerError, Result};
+use crate::apid::{
+    crypto::jwk::Jwk,
+    dsa::{ed25519::Ed25519Signer, Signer},
+};
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
 };
-use std::{collections::HashMap, sync::RwLock};
-
-#[derive(thiserror::Error, Debug, Clone, PartialEq)]
-pub enum KeyManagerError {
-    #[error(transparent)]
-    JwkError(#[from] JwkError),
-    #[error("Key generation failed")]
-    KeyGenerationFailed,
-    #[error("{0}")]
-    InternalKeyStoreError(String),
-    #[error("key not found {0}")]
-    KeyNotFound(String),
-}
-
-type Result<T> = std::result::Result<T, KeyManagerError>;
 
 #[derive(Default)]
 pub struct InMemoryKeyManager {
     map: RwLock<HashMap<String, Jwk>>,
 }
 
+impl Clone for InMemoryKeyManager {
+    fn clone(&self) -> Self {
+        let cloned_map = self.map.read().unwrap().clone();
+        InMemoryKeyManager {
+            map: RwLock::new(cloned_map),
+        }
+    }
+}
+
 impl InMemoryKeyManager {
     pub fn new() -> Self {
         Self {
             map: RwLock::new(HashMap::new()),
         }
     }
 
-    pub fn generate_key_material(&self) -> Result<Jwk> {
```

Comment:
so now we have to do Ed2219 ed25519_generator_generate and import private key?

Reply:
or just the old school way now, pass a key manager into bearer did

Reply:
that's right, I added some details to the [Draft RFC PR](https://github.com/TBD54566975/web5-rs/pull/240/files)

> Removal of `InMemoryKeyManager`'s `METHOD generate_key_material()`: this is nonsensical since the key manager implementation stores keys in memory and does not expose the functionality to export keys. The only use case would be for short-lived ephermal use cases, in which case, the developer can use the `Ed25519Generator` to generate the keys and call `METHOD import_jwk()` or the new `CONSTRUCTOR(private_jwks: []Jwk)`.

We're moving fast to unblock development, but I'll still backfill reasoning in the RFCs

We'll probably still want to expose a generator function, but I realized I was getting ahead of myself with convenience things, so removing for now